necrobot/match/cmd_match.py

```diff
@@ -741,23 +741,28 @@ def __init__(self, bot_channel):
     # Add/delete the cawmentary
     if add:
         match.set_cawmentator_id(author_user.user_id)
-        await NEDispatch().publish(event_type='set_cawmentary', match=match)
+    else:
+        match.set_cawmentator_id(None)
+
+    # Add/remove the cawmentator from the race room
+    if !match.is_self_cawmentated and author_user.member is not None:
+        if add:
+            await cmd.channel.set_permissions(author_user.member,
+                read_messages=True, read_message_history=False, send_messages=False,
```

Comment:
I added `read_message_history=False` b/c of privacy concerns; this might be more acceptable? but possibly still not good enough; lmk